### ISSUE TITLE
homebrew tap for cyclone

### ISSUE THREAD

**Issue by adamfeuer** [justinethier/cyclone#316]
@justinethier Here's a partially-working homebrew tap for cyclone: https://github.com/adamfeuer/homebrew-cyclone/blob/master/cyclone.rb

Looks like there will be some changes needed to the cyclone repo as well. I'll work on this over the next few days and submit a PR when I get it working.

Right now the homebrew formula compiles and installs cyclone-bootstrap ok, and I can use it to compile cyclone, but then when I compile cyclone with cyclone, that fails on linking... so more Makefile stuff in cyclone is probably needed.

**Comment by justinethier** [comment]
@adamfeuer - Great, thanks! I did merge the `Makefile.config` changes back to this repo but did not modify anything in `Makefile`. So those changes probably need to be migrated over as well for the Mac build.

**Comment by adamfeuer** [comment]
@justinethier Thanks for merging the `Makefile.config` changes into cyclone-bootstrap! I did make a local branch of cyclone with those changes, but that wasn't enough to get cyclone to compile itself. I think some other changes are needed. I'll submit a PR to when I get that working.

**Comment by adamfeuer** [comment]
@justinethier I was able to get cyclone to compile itself with some pretty minor changes to the `Makefile` and `Makefile.config`: 

[compare justinethier/cyclone/master to adamfeuer/cyclone/homebrew-compatibility](https://github.com/justinethier/cyclone/compare/master...adamfeuer:homebrew-compatibility)

I'm not going to submit a PR just yet because I want to test on Linux and also see if I need other changes to make a homebrew tap to work. 

homebrew taps use release artifacts to build– is there a way for me to make a private release using my own fork and branch so I can test the homebrew tap before submitting the PR? Do you have a script that packages the release? I looked around in the repo but didn't see it. Or do you use the Gitub web UI to do it?

**Comment by justinethier** [comment]
@adamfeuer - Sounds great, thanks!

When making a release I tag and push that to Github. Github generates the zip/tar.gz packages (presumably using their own internal scripts) and creates download links for them. Do you think you can create your own archive and just use a link to that to test your tap?

**Comment by adamfeuer** [comment]
@justinethier Ok, these modifications do also work on linux.

**Comment by adamfeuer** [comment]
@justinethier Thanks for the info on the releases... I'll try to do a tag (that is clearly a testing tag) on my fork and see if that generates a release there. I'll delete it after I finish the test to prevent confusion.

**Comment by adamfeuer** [comment]
@justinethier I got the homebrew tap to work for cyclone-bootstrap and for cyclone. It was just easier to make two separate formulas, even though there is some code duplication.

I submitted #317 that makes the changes necessary for the homebrew taps to work. Once this is merged and releases are made in cyclone and cyclone bootstrap, I can update the homebrew formulas to point at the latest releases and submit a PR for updating the macOS install documentation. This will make it easier for Mac users to install cyclone.

It would be really nice to have some continuous integration that would update the homebrew formula when new releases are made. I'll see if I can get this to work from my github/travis account, if not we could do something from your account (or go that way anyway if you want).

Do you have a Debian/Ubuntu PPA for debian packages? If not, and if you want that, I'll look into setting that up too.

**Comment by justinethier** [comment]
Thanks @adamfeuer ! This has been merged, let me see if I can get a 0.11 release pushed out either this weekend or early next week. I appreciate all your work on this. Please let me know how you make out with CI.

I do not have a PPA but am certainly interested. That would be web-based, correct? Would it make sense to host on github pages alongside with the Cyclone website?

**Comment by adamfeuer** [comment]
@justinethier You're welcome. I will definitely give the homebrew tap CI a shot next week, it should be straightforward. I'd like myself and more other people to be able to easily install cyclone on macOS.

As for the Ubuntu PPA... I think it would be hosted on Ubuntu's site, and they build the actual deb binaries, but something has to trigger it after a new release... after looking into it I am not sure I'm up to that right now. But I will keep thinking about it. There are other people who have made scripts to automatically update Ubuntu PPAs so maybe we can adapt one of them.

**Comment by justinethier** [comment]
@adamfeuer - A new release (0.11) has just been made, so you are good to go for updating the homebrew formulas when you have a chance. Thanks again for all your help!

**Comment by adamfeuer** [comment]
@justinethier I wrote a script to update the homebrew formulas, and uploaded the new formulas to the homebrew tap repo. Installation of cyclone is much easier on Mac now. 

I submitted [cyclone-bootstrap PR#9](https://github.com/justinethier/cyclone-bootstrap/pull/9) and [cyclone PR#318](https://github.com/justinethier/cyclone/pull/318) to update the Mac install instructions.

The script to update the homebrew formula is in Python and is quick and dirty: [update-formulas.py](https://github.com/adamfeuer/homebrew-cyclone/blob/master/update-formulas.py). I'll try to get that running in travis CI on the weekend or early next week. I have to figure out how to get the script to cleanly cut a branch, commit to it, merge it, and push the changes first.

At some point this would be good to rewrite in Cyclone Scheme. Maybe I'll do that after I get an Ubuntu PPA working.

**Comment by justinethier** [comment]
@adamfeuer This is great, thanks!

Is it too late to use a repository under **https://github.com/cyclone-scheme** to host the tap? Or would that not make sense?

**Comment by adamfeuer** [comment]
@justinethier You're welcome. 

Re: using a repo like https://github.com/justinethier/cyclone-scheme - that's easy. Nothing has been announced so all we need to do is fork the repo into your account and update the instructions. If you want me to maintain it (which would be great since you don't have a Mac? 😉 ) then you would give me commit rights to it.

Or did you mean you want to set up a separate cyclone-scheme account that's not your personal one? If so, same for that too. Just let me know.

**Comment by justinethier** [comment]
Well, there is a separate cyclone-scheme organization set up over here: https://github.com/cyclone-scheme . The thought was that if the homebrew repo was created over there then everyone could access it as needed.

I tried sending you an invitation - did you receive it? If not I can try sending again. Also, if you need to have more permissions just let me know.

Longer-term the cyclone repos should probably be moved to the organization as well... but for the time being they will remain where they are, since there are already several external links to them.

**Comment by adamfeuer** [comment]
@justinethier Ok, new [homebrew-cyclone repo](https://github.com/cyclone-scheme/homebrew-cyclone) created and PRs submitted to update the installation instructions.

**Comment by adamfeuer** [comment]
@justinethier Also, thanks for the invite to the cyclone-scheme organization!

**Comment by justinethier** [comment]
@adamfeuer - Awesome, thank you so much!

**Comment by justinethier** [comment]
@adamfeuer - No problem :)

**Comment by justinethier** [comment]
@adamfeuer - Thanks for making this happen! I am going to close the issue as this has been available for some time now.
